pkg/objectstate/rte/nodegroup.go

```diff
@@ -74,6 +82,8 @@ func (obj nodeGroupFinder) FindState(mf Manifests, tree nodegroupv1.Tree) []obje
 
 	desiredDaemonSet := mf.Core.DaemonSet.DeepCopy()
 	desiredDaemonSet.Name = generatedName
+	// it's possible that the hash will be empty if kubelet controller hasn't created a configmap
+	rteupdate.DaemonSetHashAnnotation(desiredDaemonSet, existingDaemonSet.rteComputedConfigHash)
```

Comment:
I'm thinking aloud. Should we set it to "" (empty string) or should we set it only if `!= ""`? What are the pros and cons in both scenarios?

Reply:
do we log when we change annotation? we probably should (V>=4)

Reply:
if kubeletconfig has been completely deleted from the cluster, we should be able to signal that correctly.
setting the annotation value to `""` gives us what we need.
It will cause RTE pods to restart and they'll reflect the right behavior as in the case of no kubeletconfig configuration at all.

Reply:
good point, it's worthy to keep it. Please do *NOT* resubmit just to add logs